test_dag_order/helpers.py

```diff
@@ -0,0 +1,220 @@
+
+    def __len__(self):
+        return len(self.fixtures)
+
+
+class CollectionToTest(Collection):
+    """A Collection containing a rules directory"""
+
+    def __init__(self, path: Union[str, pathlib.Path]):
+        super().__init__(path)
+
+    @property
+    def rules_directory(self) -> Path:
+        """Path to the rules directory"""
+        return self.path / RULES_DIRECTORY_NAME
+
+    def get_rules(self) -> Generator[_Rule, None, None]:
+        """Yields every rules from the rules directory"""
+        if not self.rules_directory.exists():
+            logger.warning(
+                f"Rules directory {self.rules_directory} does not exists for",
+                "collection {self.name}, skipping",
+            )
+            return
+        logger.info(f"Loading rules for collection {self.name}")
+        for rules_file_path in self.rules_directory.glob("*" + YML_EXTENSION):
+            logger.debug(f"Loading rules from {rules_file_path}")
+            with open(rules_file_path) as f:
+                rules: dict = yaml.safe_load(f)
+                for [source, rule_dict] in rules.items():
+                    yield _Rule.from_rule_dict(
+                        source=source, rules=rule_dict, rules_file_path=rules_file_path
+                    )
+
+
+class append_collection_list_action(argparse.Action):
+    """argpase action to create a list of CollectionToTest from paths"""
+
+    def __call__(self, parser, namespace, values: pathlib.Path, option_string=None):
+        # Initialize the attribute if it doesn't exist
+        if not hasattr(namespace, self.dest) or getattr(namespace, self.dest) is None:
+            setattr(namespace, self.dest, [])
+        # Append the path to the namespace
+        try:
+            collection = CollectionToTest(values)
+            collection._inventory_reader = _MockInventoryReader(["localhost"])
+            getattr(namespace, self.dest).append(collection)
+        except (
+            PathDoesNotExistsError,
+            PathIsNotADirectoryError,
+            MissingMandatoryDirectoryError,
+        ) as e:
+            parser.error(f"{e}")
+
+
+def resolve_components(
+    service_components: Iterable[str], collections: Collections
+) -> tuple[frozenset[str], dict[str, str]]:
+    """Resolve the components from the service components.
+
+    If a service is given, all the components of the service are returned.
+
+    Args:
+        service_components: Iterable of service components.
+        collections: Collections to use to resolve the components.
+
+    Returns:
+        Tuple of the resolved components and the service component map.
+
+    Examples:
+        >>> resolve_components(["service1", "service2"])
+        (frozenset({"service1", "service2"}), {})
+        >>> resolve_components(["service1", "service2_component1"])
+        (frozenset({"service1", "service2_component1"}), {"service2_component1": "service2"})
+        >>> resolve_components(["service1", "service2"])
```

Comment:
The third example is the same as the first and the fourth example is the same as the second right?